optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintStream.java

```diff
@@ -124,7 +124,7 @@ default Constraint penalize(String constraintName, Score<?> constraintWeight) {
      * <p>
      * The constraintWeight comes from an {@link ConstraintWeight} annotated member on the {@link ConstraintConfiguration},
      * so end users can change the constraint weights dynamically.
-     * This constraint is deactivated if the {@link ConstraintWeight} is zero.
+     * This constraint may be deactivated if the {@link ConstraintWeight} is zero.
```

Comment:
This is changing the javadocs to the implementation reality (of CS-D), today.

We should investigate the work for the reverse: change the implementation reality to adhere to the javadocs :)
In theory, it's matter of deactiving rules if the constraintWeight is zero. We should reopen that discussion with Mario for `Rule.setEnabled(false)`.

Reply:
I'm OK with talking to Mario. Yet, the Javadocs - in reality - are lying to users now. :-)

Reply:
There is another option. We could rebuild the KieBase every time we create a new score director.
Since there isn't any compilation (we're doing straight executable model), it shouldn't be too slow.